### PR TITLE
Support customize load balancer health probe protocol and request path

### LINKED ISSUE #76657: Support additional health probe types for Azure Load Balancer
**What would you like to be added**: Support for additional load balancer health probe configurations (specifically HTTP and HTTPS) for Azure Load Balancers.

**Why is this needed**: The generated Azure Load Balancers for `LoadBalancer` type services only support TCP health probes. Those guarantee that the backend pool consists of nodes listening on the target port but do not validate whether or not anything is responding. We've also noticed elevated volumes  SSL handshake errors generated by the TCP health checks targeting 443 when using more verbose log level configurations in ingress-nginx.

It would be valuable to configure the type of health probe used by the AKS-created Azure Load Balancer.

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Currently, the protocol of the health probe is defined as below:
1. for local services, HTTP and `/healthz` would be used;
2. for cluster TCP services, TCP would be used;
3. for cluster UDP services, no health probes.
Here we introduce two service annotations `service.beta.kubernetes.io/azure-load-balancer-health-probe-protocol` and `service.beta.kubernetes.io/azure-load-balancer-health-probe-request-path`, which determine the new health probe behavior. If the `service.beta.kubernetes.io/azure-load-balancer-health-probe-protocol` is set, both local and cluster TCP services would use the specified health probe protocol. If the `service.beta.kubernetes.io/azure-load-balancer-health-probe-request-path` is set, the specified request path would be used instead of `/healthz`. Note that the request path would be ignored when using TCP or the `service.beta.kubernetes.io/azure-load-balancer-health-probe-protocol` is empty.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/76657

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Support customize load balancer health probe protocol and request path
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```
/sig cloud-provider
/area provider/azure
/priority important-soon
/triage accepted

### CODE REVIEW

**Comment by nilo19** [comment]
The azure deployment is failed: `error creating cluster: cannot deploy: cannot get the create deployment future response: Future#WaitForCompletion: context has been cancelled: StatusCode=200 -- Original Error: context deadline exceeded`

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure.go [hunk 1]
```diff
@@ -645,6 +653,16 @@ func parseConfig(configReader io.Reader) (*Config, error) {
 		return nil, err
 	}
 
+	if config.LoadBalancerHealthProbeProtocol == "" && config.LoadBalancerHealthProbeRequestPath != "" {
+		klog.Warningf("the LoadBalancerHealthProbeProtocol is not set, would ignore the LoadBalancerHealthProbeRequestPath.")
+		config.LoadBalancerHealthProbeRequestPath = ""
+	}
+	if strings.EqualFold(config.LoadBalancerHealthProbeProtocol, string(network.ProbeProtocolTCP)) {
```
Comment (eric-isakson) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure.go:660]: Shouldn't this only warn if  && config.LoadBalancerHealthProbeRequestPath != ""

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve
/retest

**Pull request merged**